branchLayout/impl/src/test/java/com/virtuslab/branchlayout/unit/BranchLayoutTestSuite.java

```diff
@@ -49,6 +50,78 @@ public void withBranchSlideOut_givenNonRootExistingBranch_slidesOut() {
+    String parentName0 = "parent0";
+    String childName0 = "child0";
+    String childName1 = "child1";
+    String parentName1 = "parent1";
+    String childName2 = "child2";
+
+    /*-
+        root                          root
+          parent0
+            child0                      child0
+            child1                      child1
+          parent1
+            child2
+    */
+
+    List<IBranchLayoutEntry> childBranches0 = List.of(
+        new BranchLayoutEntry(childName0, /* customAnnotation */ null, List.empty()),
+        new BranchLayoutEntry(childName1, /* customAnnotation */ null, List.empty()));
+
+    val entry0 = new BranchLayoutEntry(parentName0, /* customAnnotation */ null, childBranches0);
+
+    List<IBranchLayoutEntry> childBranches1 = List.of(
+        new BranchLayoutEntry(childName2, /* customAnnotation */ null, List.empty()));
+
+    val entry1 = new BranchLayoutEntry(parentName1, /* customAnnotation */ null, childBranches1);
+    val rootEntry = new BranchLayoutEntry(rootName, /* customAnnotation */ null, List.of(entry0, entry1));
+    val branchLayout = new BranchLayout(List.of(rootEntry));
+    //then
+    assertEquals(branchLayout.findNextEntry(childName1).getName(), parentName1);
+    assertEquals(branchLayout.findPreviousEntry(childName1).getName(), childName0);
+    assertEquals(branchLayout.findNextEntry(parentName0).getName(), childName0);
+    assertEquals(branchLayout.findPreviousEntry(childName0).getName(), parentName0);
+    assertEquals(branchLayout.findPreviousEntry(parentName1).getName(), childName1);
+    assertNull(branchLayout.findPreviousEntry(rootName));
+    assertNull(branchLayout.findNextEntry(childName2));
+  }
+
+  @Test
+  public void givenSingleEntryLayout_nextAndPreviousShouldBeNull() {
+    // given
+    String rootName = "root";
+
+    val rootEntry = new BranchLayoutEntry(rootName, /* customAnnotation */ null, List.empty());
+    val branchLayout = new BranchLayout(List.of(rootEntry));
+    //then
+    assertNull(branchLayout.findPreviousEntry(rootName));
+    assertNull(branchLayout.findNextEntry(rootName));
+  }
+
+  @Test
+  public void givenOnlyRootsEntryLayout_nextAndPreviousShouldBeCalculated() {
+    // given
+    String rootName = "root";
+    String rootName1 = "root1";
+    String rootName2 = "root2";
+
+    val rootEntry = new BranchLayoutEntry(rootName, /* customAnnotation */ null, List.empty());
+    val rootEntry1 = new BranchLayoutEntry(rootName1, /* customAnnotation */ null, List.empty());
+    val rootEntry2 = new BranchLayoutEntry(rootName2, /* customAnnotation */ null, List.empty());
+    val branchLayout = new BranchLayout(List.of(rootEntry, rootEntry1, rootEntry2));
+
+    //then
+    assertNull(branchLayout.findPreviousEntry(rootName));
+    assertNull(branchLayout.findNextEntry(rootName2));
+    assertEquals(branchLayout.findNextEntry(rootName1).getName(), rootName2);
+    assertEquals(branchLayout.findPreviousEntry(rootName1).getName(), rootName);
```

Comment:
@mkondratek this is the test

Reply:
That's a slightly different case though... what does `branchLayout.findPreviousEntry(rootName2)` return?

Reply:
Similarly, I'd also add a check for `branchLayout.findNextEntry(rootName)`

Reply:
hmmm 🤔 that's not exactly the case I am worried about.

Try this one:
```
    assertEquals(branchLayout.findPreviousEntry(rootName2).getName(), rootName1);
```
I think it should pass but it fails

Reply:
but without `- 1` it works correctly

Reply:
Sorry, that was a mistake in the array bound which is now corrected . Now I also added the test case and everything seems to work.

Reply:
Good spot @mkondratek